pkg/cmd/component/environment/add.go

```
@@ -0,0 +1,234 @@
+			if !exists {
+				log.Progressf("the Application : %s doesn't exists in Path %s", io.ApplicationName, io.Output)
+				io.ApplicationName = ui.SelectApplicationNameComp(ApplicationNameAddEnvironmentMessage)
+			}
+		}
+	} else {
+		io.ApplicationName = ui.SelectApplicationNameComp(ApplicationNameAddEnvironmentMessage)
+	}
+	ui.AppNameGiven = io.ApplicationName
+	presentComponents := ui.NumberOfComponents(filepath.Join(io.Output, io.ApplicationName, "components"))
+	if len(presentComponents) == 0 {
+		return fmt.Errorf("there are no components in the %s application in folder: %s ", io.ApplicationName, io.Output)
+	} else {
+		if io.ComponentName != "" {
+			err := ui.ValidateName(io.ComponentName)
+			if err != nil {
+				log.Progressf("%v", err)
+				io.ComponentName = ui.SelectComponentNameDelete(filepath.Join(io.Output, io.ApplicationName, "components"))
+			}
+			exists, _ := ioutils.IsExisting(appFs, filepath.Join(io.Output, io.ApplicationName, "components", io.ComponentName))
+			if !exists {
+				log.Errorf("the component : %s does not exists in Application : %s at %s ", io.ComponentName, io.ApplicationName, io.Output)
+				io.ComponentName = ui.SelectComponentNameDelete(filepath.Join(io.Output, io.ApplicationName, "components"))
+			}
+		} else {
+			io.ComponentName = ui.SelectComponentNameDelete(filepath.Join(io.Output, io.ApplicationName, "components"))
+		}
+	}
+	ui.ComponentNameGiven = io.ComponentName
+	if io.EnvironmentName != "" {
+		err := ui.ValidateName(io.EnvironmentName)
+		if err != nil {
+			log.Progressf("%v", err)
+			io.EnvironmentName = ui.AddEnvironmentName()
+		}
+		log.Progressf("Path to check is %s", filepath.Join(io.Output, io.ApplicationName, "components", io.ComponentName, "overlays", io.EnvironmentName))
+		exists, _ := ioutils.IsExisting(appFs, filepath.Join(io.Output, io.ApplicationName, "components", io.ComponentName, "overlays", io.EnvironmentName))
+		if exists {
```

We need to throw an error after the flag has been passed with a valid name and the environment exist already... before we ask for new environment name.